easybuild/tools/options.py

```diff
@@ -2011,14 +2012,19 @@ def opts_dict_to_eb_opts(args_dict):
             prefix = '--'
         option = prefix + str(arg)
         value = args_dict[arg]
-        if isinstance(value, (list, tuple)):
-            value = ','.join(str(x) for x in value)
 
-        if value in [True, None]:
+        if str(arg) in allow_multiple_calls:
+            if not isinstance(value, (list, tuple)):
+                value = [value]
+            for v in value:
+                args.append(option + '=' + str(v))
```

Comment:
We avoid using singe-letter variables outside of list comprehensions, so:

```suggestion
            args.extend(option + '=' + str(x) for x in value)